recipe_scrapers/marleyspoon.py

```diff
@@ -0,0 +1,146 @@
+
+    def title(self):
+        return normalize_string(
+            self.soup.find("h3", {"class": "recipe-title"}).get_text()
+        )
+
+    def _specs(self):
+        vocab = {
+            "de": {
+                "time": "Dauer",
+                "category": "Schwierigkeitsgrad",
+                "nutrients": "Nährwertangaben",
+            },
+            "nl": {
+                "time": "Bereidingstijd",
+                "category": "Niveau",
+                "nutrients": "Voedingswaarde",
+            },
+            "en": {
+                "time": "Serving Time",
+                "category": "Level",
+                "nutrients": "Nutrition",
+            },
+        }
+
+        self.locale = self.language()[:2]
+        if self.locale not in vocab.keys():
+            raise UnsupportedLocale(self.locale)
+
+        recipe_specs = self.soup.find("ul", {"class": "recipe-specs"})
+
+        time_label = recipe_specs.find("label", string=vocab[self.locale]["time"])
+        category_label = recipe_specs.find(
+            "label", string=vocab[self.locale]["category"]
+        )
+        nutrients_pattern = "^{}.*$".format(vocab[self.locale]["nutrients"])
+        nutrients_label = recipe_specs.find(
+            "label", string=re.compile(nutrients_pattern)
+        )
+
+        time_info = time_label.find_next_sibling("p").get_text()
+        category_info = category_label.find_next_sibling("p").get_text()
+        nutrients_info = nutrients_label.find_next_sibling("p").get_text()
+
+        return time_info, category_info, nutrients_info
+
+    def total_time(self):
+        return get_minutes(self._specs()[0])
+
+    def category(self):
+        return normalize_string(self._specs()[1])
+
+    def nutrients(self):
+        nutrients = {}
+
+        vocab = {
+            "calories": {
+                "de": "Kalorien",
+                "nl": "Calorieën",
+                "en": "Calories",
+                "unit": "calories",
+            },
+            "fatContent": {"de": "Fett", "nl": "Vet", "en": "Fat", "unit": "grams fat"},
+            "proteinContent": {
+                "de": "Eiweiß",
+                "nl": "Eiwit",
+                "en": "Proteins",
+                "unit": "grams protein",
+            },
+            "carbohydrateContent": {
+                "de": "Kohlenhydrate",
+                "nl": "Koolhydraten",
+                "en": "Carbs",
+                "unit": "grams carbohydrates",
+            },
+        }
+
+        nutrients_info = self._specs()[2].split(",")
+        for nutrient in nutrients_info:
+            for key, value in vocab.items():
+                if self.locale not in value.keys():
+                    raise UnsupportedLocale(self.locale)
+                if value[self.locale] in nutrient:
+                    nutrients[key] = (
+                        re.search(r"[\d.]+", nutrient).group() + " " + value["unit"]
+                    )
```

Comment:
It could be worth expanding this assignment onto two or more lines of code; it's slightly difficult to determine what this is doing.